quinn-proto/src/connection/mod.rs

```diff
@@ -1011,27 +1011,15 @@ impl Connection {
 
     /// Indicate what types of frames are ready to send for the given space
     fn space_can_send(&self, space_id: SpaceId, frame_space_1rtt: usize) -> SendableFrames {
-        if self.spaces[space_id].crypto.is_some() {
-            let can_send = self.spaces[space_id].can_send(&self.streams);
-            if !can_send.is_empty() {
-                return can_send;
-            }
-        }
-
-        if space_id != SpaceId::Data {
-            return SendableFrames::empty();
-        }
-
-        if self.spaces[space_id].crypto.is_some() && self.can_send_1rtt(frame_space_1rtt) {
-            return SendableFrames {
-                other: true,
-                acks: false,
-            };
-        }
-
-        if self.zero_rtt_crypto.is_some() && self.side.is_client() {
+        if self.spaces[space_id].crypto.is_some()
```

Comment:
This commit has too many steps coalesced for me to easily review, do you mind splitting it up so that we can both review more carefully for avoiding semantic changes? In particular, is `other: true, acks: false` the same as `empty()`?

Reply:
> do you mind splitting it up

Good thought, done.

> In particular, is other: true, acks: false the same as empty()?

No. In the original logic, if `acks` should be true then we would have returned from the first `if`, and `can_send_1rtt` in `Data` space always means `other` should be true.